libs/src/sdk/api/developer-apps/DeveloperAppsApi.ts

```diff
@@ -30,40 +33,53 @@ export class DeveloperAppsApi extends GeneratedDeveloperAppsApi {
   /**
    * Create a developer app
    */
-  async createDeveloperApp(requestParameters: CreateDeveloperAppRequest) {
-    const { name, userId, isPersonalAccess } = requestParameters
-    // TODO (nkang + seabass): Use Zod for validation
-    const metadataMissingValues = objectMissingValues(
-      requestParameters,
-      CREATE_DEVELOPER_APP_REQUIRED_VALUES
-    )
-    if (metadataMissingValues?.length) {
-      throw new Error(
-        `Create Developer App error - Metadata object is missing values: ${metadataMissingValues}`
-      )
-    }
+  async createDeveloperApp(
+    requestParameters: CreateDeveloperAppRequest,
+    writeOptions?: WriteOptions
+  ) {
+    const { name, userId, description } = parseRequestParameters(
+      'createDeveloperApp',
+      CreateDeveloperAppSchema
+    )(requestParameters)
 
-    const decodedUserId = decodeHashId(userId)
-    if (decodedUserId == null) {
-      throw new Error(`Create Developer App error - user id is invalid`)
-    }
     const apiSecretRaw = secp.utils.randomPrivateKey()
-    const apiKeyRaw = secp.getPublicKey(apiSecretRaw, true)
+    const walletPubKey = secp.getPublicKey(
+      apiSecretRaw,
+      /** compressed = */ false
+    )
+    const apiKeyRaw = pubToAddress(Buffer.from(walletPubKey))
+
     const apiSecret = Buffer.from(apiSecretRaw).toString('hex')
-    const apiKey = Buffer.from(apiKeyRaw).toString('hex')
+    const apiKey = apiKeyRaw.toString('hex')
+
+    const unixTs = Math.round(new Date().getTime() / 1000) // current unix timestamp (sec)
+    const message = `Creating Audius developer app at ${unixTs}`
+    const signature = await secp.sign(
+      keccak_256(message),
+      Buffer.from(apiSecretRaw),
+      {
+        recovered: true,
+        der: false
+      }
+    )
+
     const response = await this.entityManager.manageEntity({
-      userId: decodedUserId,
+      userId,
       entityType: EntityType.DEVELOPER_APP,
       entityId: 0, // Contract requires uint, but we don't actually need this field for this action. Just use 0.
       action: Action.CREATE,
       metadata: JSON.stringify({
         name,
-        address: apiKey,
-        is_personal_access: isPersonalAccess ?? false
+        description,
+        app_signature: {
```

Comment:
Context for this change: https://github.com/AudiusProject/audius-protocol/pull/5391/files

Basically we want the user to prove ownership of the developer app wallet which is why we have them send this signature upon developer app creation. Otherwise they could just use any ol' address.